### PR TITLE
Merge revamp2

### PR DESCRIPTION
**Pull request by arturum1**
Same as PR https://github.com/IObundle/iob-soc/pull/934, but its easier for me to review changes compared to if_gen2 branch

### CODE REVIEW

File: submodules/BOOTROM/Makefile
Comment (arturum1) [inline submodules/BOOTROM/Makefile:1]: This file seems outdated. Can probably be deleted.

File: submodules/BOOTROM/default.nix
Comment (arturum1) [inline submodules/BOOTROM/default.nix:1]: We don't use symlinks for default.nix anymore.
This default.nix should either be deleted, or use an include:
```
{ pkgs ? import <nixpkgs> {} }:
import ../../lib/scripts/default.nix { inherit pkgs; }
```

**Review by arturum1 — COMMENTED** [review]
I think we should implement the following changes:

File: hardware/fpga/vivado/aes_ku040_db_g/iob_soc_ku040_wrapper/iob_soc_ku040_wrapper.py [hunk 1]
```diff
@@ -115,10 +164,34 @@ def setup(py_params_dict):
...
+            fpga_wrapper_wires.append(wire)
+
+    # Get all IOb-SoC AXI interfaces
+    axi_wires = []
+    for wire in fpga_wrapper_wires:
+        # Skip non-AXI wires
+        if "interface" not in wire or wire["interface"]["type"] != "axi":
+            continue
+        axi_wires.append(wire)
+
```
[the remark this answers is not in the document]
Comment (jjts) [reply hardware/fpga/vivado/aes_ku040_db_g/iob_soc_ku040_wrapper/iob_soc_ku040_wrapper.py:190]: Not now.

Maybe iob_soc.py can become a tool to create socs in py2 but not in this repo. In that case this local iob_soc.py script  will be even simpler

File: iob_soc.py [hunk 2]
```diff
@@ -641,13 +561,57 @@ def setup(py_params_dict):
...
+) reset_pulse (
+    .clk_i  (clk_i),
+    .arst_i (arst_i),
+    .cke_i  (cke_i),
+    .start_i(cpu_rst_start_pulse),
+    .pulse_o(cpu_reset)
+);
+            """,
+        },
+    ]
```
[the remark this answers is not in the document]
Comment (jjts) [reply iob_soc.py:614]: we no longer have pusle gen to generate reset . It can be removed and become like ifgen2

File: submodules/BOOTROM/Makefile
Comment (jjts) [reply submodules/BOOTROM/Makefile:1]: yes

File: submodules/BOOTROM/default.nix
Comment (jjts) [reply submodules/BOOTROM/default.nix:1]: right

File: hardware/fpga/vivado/aes_ku040_db_g/iob_soc_ku040_wrapper/iob_soc_ku040_wrapper.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (arturum1) [reply hardware/fpga/vivado/aes_ku040_db_g/iob_soc_ku040_wrapper/iob_soc_ku040_wrapper.py:190]: Resolved in 9fa87fc0bbf5e59d7ffc56daed456982f0ac0318

File: Makefile [hunk 3]
```diff
@@ -35,10 +35,10 @@ fpga-run:
...
 
 fpga-test:
-	make clean setup fpga-run BOARD=cyclonev_gt_dk INIT_MEM=1 USE_EXTMEM=0 
-	make clean setup fpga-run BOARD=cyclonev_gt_dk INIT_MEM=0 USE_EXTMEM=1 
-	make clean setup fpga-run BOARD=aes_ku040_db_g INIT_MEM=1 USE_EXTMEM=0 
-	make clean setup fpga-run BOARD=aes_ku040_db_g INIT_MEM=0 USE_EXTMEM=1 
+	make clean setup fpga-run BOARD=CYCLONEV-GT-DK INIT_MEM=1 USE_EXTMEM=0
+	make clean setup fpga-run BOARD=CYCLONEV-GT-DK INIT_MEM=0 USE_EXTMEM=1
+	make clean setup fpga-run BOARD=AES-KU040-DB-G INIT_MEM=1 USE_EXTMEM=0
+	make clean setup fpga-run BOARD=AES-KU040-DB-G INIT_MEM=0 USE_EXTMEM=1
```
[the remark this answers is not in the document]
Comment (arturum1) [reply Makefile:41]: Resolved in 9fa87fc0bbf5e59d7ffc56daed456982f0ac0318

File: iob_soc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (arturum1) [reply iob_soc.py:614]: Resolved in 9fa87fc0bbf5e59d7ffc56daed456982f0ac0318

**Comment by arturum1** [comment]
Changes included in PR https://github.com/IObundle/iob-soc/pull/936

**Pull request closed without merging**